hack/login_ci_cache

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+set -e
+
+if [ -z "$TRAVIS_JOB_ID" ]; then
+  exit 0
+fi
+
+repo=""
+
+if [ -n "$MASTERCACHE_PASSWORD" ]; then
+  repo="cicache.buildk.it/moby/buildkit/master"
+  echo "$MASTERCACHE_PASSWORD" | docker login -u buildkitmaster --password-stdin cicache.buildk.it
+elif [ "$TRAVIS_EVENT_TYPE" == "pull_request" ]; then
+  repo="cicache.buildk.it/moby/buildkit/pr$TRAVIS_BUILD_ID"
+  session=$(curl -F jobid=$TRAVIS_JOB_ID -F scope=moby/buildkit/pr https://cicache.buildk.it/token/newsession)
+  sessionID=$(echo $session|cut -f 1 -d ' ')
+  echo $session|cut -f 2 -d ' '
+  mkdir -p ~/.docker/
+  curl -F sessionid=$sessionID -F scope=moby/buildkit/pr https://cicache.buildk.it/token/credentials > ~/.docker/config.json
```

Comment:
Is this credential for pull-only? If not there might be a security concern?

Reply:
the credential is for a fixed scope for a specific PR, only available to the travis job. source is in https://github.com/tonistiigi/ci-token-server